docker/Dockerfile-api

```diff
@@ -22,7 +23,8 @@ RUN dnf -y install \
 COPY . .
 COPY ./docker/cachito-httpd.conf /etc/httpd/conf/httpd.conf
 
-RUN pip3 install -r requirements.txt --no-deps --no-cache-dir --require-hashes \
+RUN --mount=type=bind,source=.git,target=/src/.git,Z \
```

Comment:
Hadolint is going to have the same parsing error here on the "Z" as over in [Bruno's cachi2 Containerfile PR](https://github.com/containerbuildsystem/cachi2/pull/3#discussion_r994842242). Since this is a parsing error instead of a rule check, I don't think it can be ignored.

Should we just copy the .git directory in?

Reply:
How did these builds ever work with setuptools-git-versioning :confused: 

Yeah, I'd say we should copy it and then `rm -rf` after installing. Maybe also uninstall `git-core`

Reply:
Oh that's how
```shell
$ podman run --rm -ti quay.io/containerbuildsystem/cachito-workers:latest pip list | grep cachito
cachito            0.0.1
```

Reply:
Yes, I chuckled a little bit when I saw that yesterday :sweat_smile:

One issue I can see with removing git-core is that in the absence of a default "fallback" version, [the local development install](https://github.com/containerbuildsystem/cachito/blob/7662e45d43f74e96a1b745112688c0b4ed7b9e62/docker-compose.yml#L57) will hit an error when trying to install the cachito package: `OSError: 'git' was not found`